examples/metrics-basic/src/main.rs

```diff
@@ -43,23 +43,21 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
     );
 
     // Create a ObservableCounter instrument and register a callback that reports the measurement.
-    let observable_counter = meter
+    let _observable_counter = meter
         .u64_observable_counter("my_observable_counter")
         .with_description("My observable counter example description")
         .with_unit(Unit::new("myunit"))
+        .with_callback(|observer| {
+            observer.observe(
+                100,
+                &[
+                    KeyValue::new("mykey1", "myvalue1"),
+                    KeyValue::new("mykey2", "myvalue2"),
+                ],
+            )
+        })
         .init();
 
-    meter.register_callback(&[observable_counter.as_any()], move |observer| {
```

Comment:
From Community Meeting 5/7:

Lalit, Zhongyang:
1. Mark this under unstable flag.
2. Unregister callback is only available with this.
3. This is part of API, so higher bar to change than SDK/View/Exporter
4. Removing this and adding in future might be difficult to achieve without breaking change.

Zhongyang: Opposed to removing it, because it was part of documented API, not because it is *required*.
Suggestion: Mark as unstable, and make it just compile - so let user take the risk. (docs are showing the right way).
Okay with: If this is too complex to achieve, Zhongyang won't block from proceeding with removal.

Lalit: +1 to above.